Chat/assignment-chat/server-chat/src/main/java/ru/assignment/net/Session.java

```diff
@@ -0,0 +1,141 @@
+package ru.assignment.net;
+
+import ru.assignment.message.ChatMessage;
+import ru.assignment.model.ChatModel;
+import ru.assignment.model.ChatModelListener;
+
+import java.io.*;
+import java.net.Socket;
+import java.util.Scanner;
+
+public class Session implements Runnable, ChatModelListener {
+    private Socket socket;
+    private final int sessionIdentifier;
+    private ChatModel chatModel;
+    private Scanner scanner;
+    private PrintWriter writer;
+
+    public Session(Socket socket, int sessionIdentifier,
+                   ChatModel chatModel) throws IOException {
+        this.socket = socket;
+        this.sessionIdentifier = sessionIdentifier;
+        this.chatModel = chatModel;
+        initConfiguration();
+    }
+
+    public void run() {
+        System.out.println("Session startSession");
+        listen();
+        closeSession();
+    }
+
+    public void initConfiguration() throws IOException {
+        try {
+            scanner = new Scanner(socket.getInputStream());
+            writer = new PrintWriter(socket.getOutputStream());
+        } catch (IOException e) {
+            closeSocket();
+            throw e;
+        }
+        chatModel.addListener(this);
```

Comment:
"this" publication before object is constructed is unsafe and should be avoided.
Please google it, this is quite common mistake.